src/fshtypes/FshCodeSystem.ts

```diff
@@ -26,52 +23,6 @@ export class FshCodeSystem extends FshEntity {
     return 'FshCodeSystem';
   }
 
-  addConcept(newConcept: ConceptRule) {
-    if (this.checkConcept(newConcept)) {
-      this.rules.push(newConcept);
-    }
-  }
-
-  checkConcept(newConcept: ConceptRule, extraRules: Rule[] = []): boolean {
-    const rulesToCheck = [...this.rules, ...extraRules];
-    const existingConcept = rulesToCheck.find(
-      rule => rule instanceof ConceptRule && rule.code == newConcept.code
-    ) as ConceptRule;
-    if (existingConcept) {
-      // if the new ConceptRule has only a code and a hierarchy,
-      // and the existing ConceptRule with that code has a matching hierarchy,
-      // the user may simply be referencing the existing concept to establish context.
-      if (
-        newConcept.display == null &&
-        newConcept.definition == null &&
-        isEqual(existingConcept.hierarchy, newConcept.hierarchy)
-      ) {
-        return false;
-      } else {
-        throw new CodeSystemDuplicateCodeError(this.id, newConcept.code);
-      }
-    }
-    // check the hierarchy, if applicable
-    // for each predecessor element in the new concept's hierarchy, we should be able to find a rule that
-    // 1. defines a concept
-    // 2. with the predecessor's code
-    // 3. and a hierarchy equal to everything before it in the new concept's hierarchy
-    newConcept.hierarchy.forEach((predecessor, i) => {
-      if (
-        !rulesToCheck.some(rule => {
-          return (
-            rule instanceof ConceptRule &&
-            rule.code === predecessor &&
-            isEqual(rule.hierarchy, newConcept.hierarchy.slice(0, i))
-          );
-        })
-      ) {
-        throw new CodeSystemIncorrectHierarchyError(this.id, newConcept.code);
-      }
-    });
```

Comment:
Is there somewhere that needs to be doing this check still? Or is there a reason it isn't needed anymore?

Reply:
The equivalent logic is in `CodeSystemExporter.setConcepts`. The loop that iterates over `concept.hierarchy` traverses the existing tree of concepts, and will log an error if it can't find the complete path specified in the hierarchy. The error log is followed by a return, meaning that the concept is not added to the exported CodeSystem.

Reply:
Ah okay I see. I got confused since that code wasn't specifically new and didn't have the `CodeSystemIncorrectHierarchyError` thrown, but I see they're accomplishing the same thing. In that case, do you think it's safe to get rid of the `CodeSystemIncorrectHierarchyError` class entirely? It doesn't look like it's used anywhere else.